apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/AuthenticatorManager.kt

```diff
@@ -0,0 +1,35 @@
+package com.twilio.livevideo.app.manager
+
+import com.twilio.livevideo.app.BuildConfig
+import com.twilio.livevideo.app.repository.datasource.local.LocalStorage
+
+class AuthenticatorManager(
+    private val localStorage: LocalStorage
+) {
+
+    fun storePasscode(passcode: String) {
+        localStorage.putStringData(FULL_PASSCODE_KEY, passcode)
+        localStorage.putStringData(URL_PASSCODE_KEY, extractPasscodeUrl(passcode))
+    }
+
+    fun getPasscode(): String = localStorage.getStringData(FULL_PASSCODE_KEY) ?: ""
+
+    fun getPasscodeURL(): String = localStorage.getStringData(URL_PASSCODE_KEY) ?: ""
+
+    fun isPasscodeValid(): Boolean = getPasscode().isNotEmpty()
+
+    fun extractPasscodeUrl(passcode: String): String {
+        var passcodeUrl = passcode.substring(passcode.length - 8)
+        passcodeUrl =
+            "${passcodeUrl.substring(0, 4)}-${passcodeUrl.substring(4, passcodeUrl.length)}"
+        return passcodeUrl.trim()
```

Comment:
There may be an issue with line 22. We designed the passcode so that the serverless ID on the end can be any length >= 4. And so I think if you start parsing the string from the end instead of the beginning there may be issues, if not now in the future.

You might take a look at https://github.com/twilio/twilio-live-interactive-video/blob/main/apps/ios/LiveVideo/LiveVideo/Managers/Auth/PasscodeComponents.swift to see if you like any ideas in this file. It encapsulates all of the passcode decoding in a separate file. 

Also the passcode decoding might be a good way to add some unit tests. It should be easy to test and high value. These are the tests I made for iOS: https://github.com/twilio/twilio-live-interactive-video/blob/main/apps/ios/LiveVideo/LiveVideoTests/PasscodeComponentsSpec.swift

Reply:
As per our conversation, I changed this logic to process the passcode.
Regarding the Unit test. Yes, I will be adding some unit test on a different ticket. Thanks for your tests on iOS, I think I will use that as a reference to make it on Android.